src/main/java/io/pravega/perf/PerfStats.java

```diff
@@ -202,12 +220,133 @@ private long windowTimeMS(long time) {
+        }
+
+        public void start() {
+            scheduler.scheduleAtFixedRate(this::recordSecondThroughput, 1, 1, TimeUnit.SECONDS);
+        }
+
+        public void stop() {
+            scheduler.shutdownNow();
+        }
+
+        public void recordSecondThroughput() {
+            synchronized (LOCk) {
+                recordThroughput(seconds, totalBytesInSecond);
+
+                totalBytesInSecond = 0;
+                seconds++;
+            }
+        }
+
+        public void recordThroughput(long seconds, long throughputBytes) {
+            throughputs.add(totalBytesInSecond);
+        }
+
+        private String[] getPercentiles() {
+            String[] values = new String[PERCENTILES.length];
+
+            throughputs.sort(Long::compareTo);
+
+            for (int i = 0; i < PERCENTILES.length; i++) {
+                int percentileIndex = (int) (throughputs.size() * PERCENTILES[i]);
+                long bytes = throughputs.get(percentileIndex);
+
+                values[i] = FileUtils.byteCountToDisplaySize(bytes);
+            }
+
+            return values;
+        }
+
+        public void record(int bytes) {
+            synchronized (LOCk) {
+                totalBytesInSecond +=bytes;
+            }
+            totalBytes+=bytes;
+            count++;
+        }
+
+        public void printTotal(long endTime) {
+            final double elapsed = (endTime - startTime) / 1000.0;
+            final double recsPerSec = count / elapsed;
+            String[] percs = getPercentiles();
+
+            log.info(String.format(
+                "throughput: %,d records %s, %,.3f records/sec, %d bytes record size, %s total data" +
+                    ", %s/s 50th, %s/s 75th, %s/s 95th, %s/s 99th, %s/s 99.9th, %s/s 99.99th.",
+                count, action, recsPerSec, messageSize, FileUtils.byteCountToDisplaySize(totalBytes),
+                percs[0], percs[1], percs[2], percs[3], percs[4], percs[5]));
+        }
+    }
+
+    /***
+     * Perf recorder that writes the number of bytes recorded every second to a CSV file.
+     */
+     @NotThreadSafe
+    static private class CSVThroughputWriter extends ThroughtputWriter {
+        final private String csvFile;
+        final private CSVPrinter csvPrinter;
+
+        CSVThroughputWriter(String action, int messageSize, long startTime, String csvFile) throws IOException {
+            super(action, messageSize, startTime);
+            this.csvFile = csvFile + "-throughput.csv";
+            csvPrinter = new CSVPrinter(Files.newBufferedWriter(Paths.get(csvFile)), CSVFormat.DEFAULT
+                .withHeader("second", action + " Throughput (bytes)"));
+        }
+
+        @Override
+        public void recordThroughput(long seconds, long throughputBytes) {
+            super.recordThroughput(seconds, throughputBytes);
+            try {
+                csvPrinter.printRecord(seconds, throughputBytes);
```

Comment:
The most useful data you can have in the CSV file is the unix timestamp `System.currentTimeMillis()` and the total number of bytes written. This allows the user to easily calculate throughput for any period using any type of averaging. It also allows correlating with other metrics.